test/fixtures/frame-scenario.js

```diff
@@ -3,13 +3,29 @@ const common = require('../common');
 
 function crasher(unused) {
   'use strict';
-  process.abort();  // Creates an exit frame.
-  return this;      // Force definition of |this|.
+  this.foo = arguments;  // Force adaptor frame on Node.js v12+
+  process.abort();       // Creates an exit frame.
+  return this;           // Force definition of |this|.
 }
 
-function eyecatcher() {
+// Force V8 to use an inferred name instead of saving the variable name as
+// FunctionName.
+let fnInferredName;
+fnInferredName = (() => function () {
   crasher();    // # args < # formal parameters inserts an adaptor frame.
+})();
```

Comment:
V8 is not inferring this on Node.js v12... We'll probably need to skip the check for this function on 12.

Upstream bug: https://bugs.chromium.org/p/v8/issues/detail?id=9807